src/js/InvertedIndex.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * InvertedIndex class
+ * @class
+ */
+class InvertedIndex {
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.files = [];
+  }
+
+  /**
+   * Validates file content
+   * @function
+   * @param {Array} fileContent content of uploaded file
+   * @return {Array|Boolean} valid fileContent or false if invalid file
+   */
+  static validateFile(fileContent) {
+    if (!(fileContent instanceof Array) || fileContent.length < 1) {
+      return false;
+    }
+
+    let invalid = false;
+    fileContent.forEach((book) => {
+      if (!(book.title && book.text)) {
+        invalid = true;
+        return;
+      }
+      if (book.text.length < 1) invalid = true;
+    });
+
+    if (invalid) return false;
+    return fileContent;
+  }
+
+  /**
+   * Strips out special characters and removes duplicate words
+   * @function
+   * @param {String} text text from the book
+   * @return {Array} An array of unique words in the text
+   */
+  static tokenize(text) {
+    return text.replace(/[^\w\s-]/g, '')
+      .toLowerCase().split(/\s+/)
+      .filter((word, index, collection) => collection.indexOf(word) === index);
+  }
+
+  /**
+   * Creates an index object from the file content and stores it
+   * @function
+   * @param {Array} fileContent content of uploaded file
+   * @param {String} filename name of uploaded file
+   * @return {boolean} true if file index was successfully created, else false
+   */
+  createIndex(fileContent, filename) {
+    if (this.files.indexOf(filename) !== -1) return false;
+
+    const result = { bookTitles: [], words: {}, allWords: [], filename };
+    fileContent.forEach((book, bookIndex) => {
+      result.bookTitles.push(book.title);
+      const words = InvertedIndex.tokenize(book.text);
+      words.forEach((word) => {
+        if (!result.words[word]) {
+          result.allWords.push(word);
+          result.words[word] = new Array(fileContent.length).fill(false);
+        }
+        result.words[word][bookIndex] = true;
+      });
+    });
+
+    this.files.push(filename);
+    this[filename] = result;
+    return true;
+  }
+
+  /**
+   * Returns stored index for a file given the filename
+   * @function
+   * @param {Object} filename name of the file
+   * @return {Array|Boolean} indexed object or false if file was not found
+   */
+  getIndex(filename) {
+    if (!this[filename]) return false;
```

Comment:
Seems you are repeating your self, This should be taking care of in validate file